x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/spatial/SpatialDisjoint.java

```diff
@@ -81,20 +86,30 @@ public class SpatialDisjoint extends SpatialRelatesFunction {
     )
     public SpatialDisjoint(
         Source source,
-        @Param(name = "geomA", type = { "geo_point", "cartesian_point", "geo_shape", "cartesian_shape" }, description = """
-            Expression of type `geo_point`, `cartesian_point`, `geo_shape` or `cartesian_shape`.
-            If `null`, the function returns `null`.""") Expression left,
-        @Param(name = "geomB", type = { "geo_point", "cartesian_point", "geo_shape", "cartesian_shape" }, description = """
-            Expression of type `geo_point`, `cartesian_point`, `geo_shape` or `cartesian_shape`.
-            If `null`, the function returns `null`.
-            The second parameter must also have the same coordinate system as the first.
-            This means it is not possible to combine `geo_*` and `cartesian_*` parameters.""") Expression right
+        @Param(
+            name = "geomA",
+            type = { "geo_point", "cartesian_point", "geo_shape", "cartesian_shape", "geohash", "geotile", "geohex" },
+            description = """
+                Expression that is either a geometry (`geo_point`, `cartesian_point`, `geo_shape` or `cartesian_shape`)
+                or a geo-grid value (`geohash`, `geotile`, `geohex`).
+                If `null`, the function returns `null`."""
+        ) Expression left,
+        @Param(
+            name = "geomB",
+            type = { "geo_point", "cartesian_point", "geo_shape", "cartesian_shape", "geohash", "geotile", "geohex" },
+            description = """
+                Expression that is either a geometry (`geo_point`, `cartesian_point`, `geo_shape` or `cartesian_shape`)
+                or a geo-grid value (`geohash`, `geotile`, `geohex`).
+                If `null`, the function returns `null`.
+                The second parameter must also have the same coordinate system as the first.
+                This means it is not possible to combine `geo_*` and `cartesian_*` parameters."""
+        ) Expression right
     ) {
         this(source, left, right, false, false);
     }
 
     private SpatialDisjoint(Source source, Expression left, Expression right, boolean leftDocValues, boolean rightDocValues) {
-        super(source, left, right, leftDocValues, rightDocValues);
+        super(source, left, right, leftDocValues, rightDocValues, true);
```

Comment:
As noted on Slack, I think there's a problem here. `supportsGrid` is set to false when unserializing, but true here, so we get different things locally and remotely.